src/App.js

```diff
@@ -3,13 +3,45 @@ import Header from './Header.js';
 import Main from './Main.js';
 import Footer from './Footer.js';
 import data from './data.json';
+import Modal  from 'react-bootstrap/Modal';
+
 
 class App extends React.Component {
-  render(){
+  constructor(props) {
+    super(props);
+    this.state = {
+      horns: '🦄',
+      show: true
+    }
+  }
+
+
+  handleUnhideModal = () => {
+    this.setState({
+      show: true,
+    })
+  }
+
+  handleHideModal = () => {
+    this.setState({
+      show: false
+    })
+  }
+
+
+  render() {
     return (
       <>
-        <Header />
-        <Main data={data} />
+        <Header horns={this.state.horns} />
+        <Modal centered show={this.state.show} onHide={this.handleHideModal}>
+          <Modal.Header closeButton>
+            <Modal.Title>This is a Modal</Modal.Title>
+          </Modal.Header>
+          <Modal.Body>
+          </Modal.Body>
+        </Modal>
```

Comment:
- This Modal code is what will be in your `selectedBeast`. The lab asks us to "Use the state in App to render an individual beast in a Modal in the SelectedBeast". So the `selectedBeast` will hold your Modal logic 👍